### PR TITLE
fix: browser error when downloading files

### PR DESCRIPTION
**Pull request by przemvs**
https://wearezeta.atlassian.net/browse/ACC-106

This fixing the play() functionality after update to React 18. Because with React > 18 state update is batched, that means that the initial code would not update the actual `video|audio` element in the DOM before calling the `play` function. 
Which mean that we would start to play the media even before the DOM element has an `src` set. 
Also in the future maybe we need to change this functionality for preloading of video/audio file.
So if the preload functionality will be implemented we need to change this functionality.

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

_Briefly describe the issue you have solved or implemented with this pull request. If the PR contains multiple issues, use a bullet list._

### Causes (Optional)

_Briefly describe the causes behind the issues. This could be helpful to understand the adopted solutions behind some nasty bugs or complex issues._

### Solutions

_Briefly describe the solutions you have implemented for the issues explained above._

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by tlebon — APPROVED** [review]

File: src/script/components/MessagesList/Message/ContentMessage/asset/AudioAsset.tsx [hunk 1]
```diff
@@ -75,14 +75,27 @@ const AudioAsset: React.FC<AudioAssetProps> = ({
...
       asset.status(AssetTransferState.UPLOADED);
     }
   };
 
+  useEffect(() => {
+    if (audioSrc && audioElement) {
+      const playPromise = audioElement.play();
+
+      if (playPromise !== undefined) {
+        playPromise.catch(error => {
```
Comment (thisisamir98) [inline src/script/components/MessagesList/Message/ContentMessage/asset/AudioAsset.tsx]:
```suggestion
        playPromise?.catch(error => {
```
I suggest using optional chaining so you can get rid of `if` statement

File: src/script/components/MessagesList/Message/ContentMessage/asset/VideoAsset.tsx [hunk 2]
```diff
@@ -104,6 +107,18 @@ const VideoAsset: React.FC<VideoAssetProps> = ({
...
+  useEffect(() => {
+    if (videoSrc && videoElement) {
+      const playPromise = videoElement.play();
+
+      if (playPromise !== undefined) {
+        playPromise.catch(error => {
+          console.error('Failed to load video asset ', error);
+        });
+      }
+    }
```
Comment (thisisamir98) [inline src/script/components/MessagesList/Message/ContentMessage/asset/VideoAsset.tsx]: same here
```suggestion
        playPromise?.catch(error => {
          console.error('Failed to load video asset ', error);
        });
      }
```

File: src/script/components/MessagesList/Message/ContentMessage/asset/AudioAsset.tsx (hunk 1 above)
Comment (przemvs) [reply src/script/components/MessagesList/Message/ContentMessage/asset/AudioAsset.tsx]: Fixed

File: src/script/components/MessagesList/Message/ContentMessage/asset/VideoAsset.tsx (hunk 2 above)
Comment (przemvs) [reply src/script/components/MessagesList/Message/ContentMessage/asset/VideoAsset.tsx]: Fixed

**Review by atomrc — APPROVED** [review]

**Pull request merged**